src/map.cpp

```diff
@@ -238,95 +237,96 @@ bool Map::placeCreature(const Position& centerPos, Creature* creature, bool exte
 
 	int32_t index = 0;
 	uint32_t flags = 0;
-	Item* toItem = nullptr;
+	std::shared_ptr<Item> toItem = nullptr;
 
-	Cylinder* toCylinder = tile->queryDestination(index, *creature, &toItem, flags);
-	toCylinder->internalAddThing(creature);
+	auto toThing = tile->queryDestination(index, creature, toItem, flags);
+	toThing->internalAddThing(creature);
 
-	const Position& dest = toCylinder->getPosition();
+	const Position& dest = toThing->getPosition();
 	getQTNode(dest.x, dest.y)->addCreature(creature);
 	return true;
 }
 
-void Map::moveCreature(Creature& creature, Tile& newTile, bool forceTeleport /* = false*/)
+void Map::moveCreature(std::shared_ptr<Creature> creature, std::shared_ptr<Tile> newTile,
+                       bool forceTeleport /* = false*/)
 {
-	Tile& oldTile = *creature.getTile();
+	auto oldTile = creature->getTile();
```

Comment:
Could this be `const auto`
```suggestion
	const auto oldTile = creature->getTile();
```